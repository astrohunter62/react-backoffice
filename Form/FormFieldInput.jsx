import React from 'react'
import PropTypes from 'prop-types'

import TextField from 'material-ui/TextField'

const FormFieldInput = ({
  id,
  title,
  value,
  isMultiline,
  handleChange,
  helperText,
  classNames
}) => (
  <TextField
    id={id}
    label={title}
    value={value}
    multiline={isMultiline}
    onChange={handleChange(id)}
    className={classNames.join(' ')}
    helperText={helperText}
    margin="normal"
    fullWidth
  />
)

FormFieldInput.propTypes = {
  id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  value: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.array,
    PropTypes.number
  ]).isRequired,
  isMultiline: PropTypes.bool.isRequired,
  helperText: PropTypes.string,
  classNames: PropTypes.array.isRequired,
  handleChange: PropTypes.func.isRequired,
}

FormFieldInput.defaultProps = {
  id: '',
  title: '',
  value: '',
  isMultiline: false,
  helperText: '',
  classNames: [],
  handleChange: () => {},
}

export default FormFieldInput
