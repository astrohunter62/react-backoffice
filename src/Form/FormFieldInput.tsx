import React, { Fragment } from "react";
import { TextField, InputAdornment, MenuItem } from "@material-ui/core";

type InputEndProps = {
  icon?: JSX.Element;
};

const InputEnd: React.SFC<InputEndProps> = ({ icon }) => (
  <Fragment>
    {icon && <InputAdornment position="end">{icon}</InputAdornment>}
  </Fragment>
);

InputEnd.defaultProps = {
  icon: undefined,
};

type FormFieldInputProps = {
  id: string;
  type?: string;
  title?: string;
  value?: string | any[] | number;
  defaultValue?: string | any[] | number;
  rows?: number;
  isMultiline?: boolean;
  helperText?: string;
  isRequired?: boolean;
  error?: boolean;
  className?: string;
  handleChange?: (...args: any[]) => any;
  select?: boolean;
  options?: string[];
  isDisabled?: boolean;
  iconEnd?: JSX.Element;
  onFocus?: (...args: any[]) => any;
  onBlur?: (...args: any[]) => any;
  onKeyPress?: (...args: any[]) => any;
};

const FormFieldInput: React.SFC<FormFieldInputProps> = ({
  id,
  type,
  title,
  value,
  isMultiline,
  handleChange,
  helperText,
  defaultValue,
  select,
  options,
  rows,
  isRequired,
  error,
  iconEnd,
  isDisabled,
  className,
  onFocus,
  onBlur,
  onKeyPress,
}) => (
  <TextField
    id={id}
    type={type}
    label={title}
    value={value}
    multiline={isMultiline}
    onChange={handleChange && handleChange(id)}
    helperText={helperText}
    defaultValue={defaultValue}
    select={select}
    required={isRequired}
    error={error}
    margin="normal"
    fullWidth
    rows={rows}
    disabled={isDisabled}
    onFocus={onFocus}
    onBlur={onBlur}
    onKeyPress={onKeyPress}
    InputProps={{
      endAdornment: <InputEnd icon={iconEnd} />,
    }}
    className={className}
  >
    {options
      ? options.map((option) => (
          <MenuItem value={option} key={option.toLowerCase().replace(/ /, "")}>
            {option}
          </MenuItem>
        ))
      : null}
  </TextField>
);

FormFieldInput.defaultProps = {
  type: "text",
  value: "",
  isMultiline: false,
  helperText: "",
  className: "",
  isRequired: false,
  error: false,
  rows: 1,
  select: false,
  options: [],
  isDisabled: false,
  handleChange: () => {},
  onFocus: () => {},
  onBlur: () => {},
  onKeyPress: () => {},
};

export default FormFieldInput;
