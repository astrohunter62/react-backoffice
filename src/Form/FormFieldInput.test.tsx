import React from "react";
import { shallow } from "enzyme";

import FormFieldInput from "./FormFieldInput";

it("renders correctly", () => {
  const tree = shallow(<FormFieldInput id="" />);

  expect(tree).toMatchSnapshot();
});

it("renders disabled", () => {
  const tree = shallow(<FormFieldInput id="" isDisabled />);

  expect(tree).toMatchSnapshot();
});
