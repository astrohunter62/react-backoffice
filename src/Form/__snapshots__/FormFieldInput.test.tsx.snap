// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<WithStyles(ForwardRef(TextField))
  InputProps={
    Object {
      "endAdornment": <InputEnd />,
    }
  }
  className=""
  disabled={false}
  error={false}
  fullWidth={true}
  helperText=""
  id=""
  margin="normal"
  multiline={false}
  onBlur={[Function]}
  onFocus={[Function]}
  onKeyPress={[Function]}
  required={false}
  rows={1}
  select={false}
  type="text"
  value=""
/>
`;

exports[`renders disabled 1`] = `
<WithStyles(ForwardRef(TextField))
  InputProps={
    Object {
      "endAdornment": <InputEnd />,
    }
  }
  className=""
  disabled={true}
  error={false}
  fullWidth={true}
  helperText=""
  id=""
  margin="normal"
  multiline={false}
  onBlur={[Function]}
  onFocus={[Function]}
  onKeyPress={[Function]}
  required={false}
  rows={1}
  select={false}
  type="text"
  value=""
/>
`;
